android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/NestedGeckoView.kt

```diff
@@ -98,6 +97,7 @@ open class NestedGeckoView(context: Context) : GeckoView(context), NestedScrolli
 
             // We don't care about other touch events
             MotionEvent.ACTION_UP, MotionEvent.ACTION_CANCEL -> {
+                inputResultDetail = InputResultDetail.newInstance(true)
```

Comment:
Let's add a comment at this line, that this needs to be done here and not in the `ACTION_DOWN` in order to reset the gesture state for the next gesture.